pkg/serviceaccount/jwt.go

```diff
@@ -170,14 +172,31 @@ func (j *jwtTokenAuthenticator) AuthenticateToken(ctx context.Context, tokenData
 		return nil, false, utilerrors.NewAggregate(errlist)
 	}
 
+	var auds authenticator.Audiences
+	// If the `aud` claim of the JWT is empty, then the JWT is a legacy service
+	// account token so we skip audience validation.
+	if len(public.Audience) > 0 {
```

Comment:
I'm not sure if this is the best/safest place for this yet.

Reply:
would it make more sense to plumb API audiences into the authenticator for both the legacy and modern JWT authenticator?

```go
tokenAudiences := public.Audience
if len(tokenAudiences) == 0 {
  // only apiserver audiences are allowed for legacy tokens
  tokenAudiences = apiAuds
}

requestedAudiences, ok := request.AudiencesFrom(ctx)
if !ok {
  // default to apiserver audiences
  requestedAudiences = apiAuds
}

auds = authenticator.Audiences(public.Audience).Intersect(apiAuds)
...
```